virttest/utils_misc.py

```diff
@@ -1680,7 +1680,7 @@ def get_all_node_meminfo(self):
             node_meminfo = {}
             numa_sys = kernel_interface.SysFS(meminfo_file % node,
                                               session=self.session)
-            for info in str(numa_sys.sys_fs_value).split("\n"):
+            for info in str(numa_sys.sys_fs_value.strip()).split("\n"):
```

Comment:
IIUC, strip() only removes spaces, the '/n' still remains. Thus `info` could be "" when there is a blank line, the issue still exists. That's my guess, I don't have a system having the blank line in meminfo. Had better to provide the test result on the system where the issue is reproduced.

Reply:
You can refer the following test: 
```
>>> file = '/sys/devices/system/node/node1/meminfo'
>>> data = os.popen('cat %s' %file).read()
>>> data.split('\n')[-1]
''
>>> data.strip().split('\n')[-1]
'Node 1 HugePages_Surp:      0'
```

I update the commit info to explain this more precisely. To test it, i call function `get_all_node_meminfo` in my case, can get the expect mem info:
```
05:18:32 DEBUG| [stdout] ====get all node mem info: {0: {'MemTotal': '2794152', 'MemFree': '2255184', 'MemUsed': '538968', 'Active': '150332', 'Inactive': '238836', 'Active(anon)': '112792', 'Inactive(anon)': '8420', 'Active(file)': '37540', 'Inactive(file)': '230416', 'Unevictable': '4224', 'Mlocked': '0', 'Dirty': '400', 'Writeback': '0', 'FilePages': '280952', 'Mapped': '113860', 'AnonPages': '106452', 'Shmem': '8772', 'KernelStack': '4344', 'PageTables': '9972', 'NFS_Unstable': '0', 'Bounce': '0', 'WritebackTmp': '0', 'KReclaimable': '28336', 'Slab': '94752', 'SReclaimable': '28336', 'SUnreclaim': '66416', 'AnonHugePages': '20480', 'ShmemHugePages': '0', 'ShmemPmdMapped': '0', 'HugePages_Total': '0', 'HugePages_Free': '0', 'HugePages_Surp': '0'}, 1: {'MemTotal': '1030376', 'MemFree': '701672', 'MemUsed': '328704', 'Active': '148140', 'Inactive': '91944', 'Active(anon)': '107240', 'Inactive(anon)': '7700', 'Active(file)': '40900', 'Inactive(file)': '84244', 'Unevictable': '4228', 'Mlocked': '0', 'Dirty': '76', 'Writeback': '0', 'FilePages': '138108', 'Mapped': '62268', 'AnonPages': '104364', 'Shmem': '8684', 'KernelStack': '1752', 'PageTables': '5140', 'NFS_Unstable': '0', 'Bounce': '0', 'WritebackTmp': '0', 'KReclaimable': '21448', 'Slab': '53424', 'SReclaimable': '21448', 'SUnreclaim': '31976', 'AnonHugePages': '8192', 'ShmemHugePages': '0', 'ShmemPmdMapped': '0', 'HugePages_Total': '0', 'HugePages_Free': '0', 'HugePages_Surp': '0'}}
```

Reply:
I see. So the blank line is always the last line? If so, this fix is valid.

Reply:
> I see. So the blank line is always the last line? If so, this fix is valid.


Yes, it is always the last line.